src/inference/src/core.cpp

```diff
@@ -51,9 +49,15 @@ Core::Core(const std::string& xml_config_file) {
 #ifdef OPENVINO_STATIC_LIBRARY
     OV_CORE_CALL_STATEMENT(_impl->register_plugins_in_registry(::getStaticPluginsRegistry());)
 #else
-    OV_CORE_CALL_STATEMENT(
-        // If XML is default, load default plugins by absolute paths
-        _impl->register_plugins_in_registry(findPluginXML(xml_config_file), xml_config_file.empty());)
+    std::string xmlConfigFile = ov::findPluginXML(xml_config_file);
+    if (xmlConfigFile.empty())
+        OV_CORE_CALL_STATEMENT(
+            // XML not found? use compile-time configuration
+            _impl->register_compile_time_plugins();)
+    else
+        OV_CORE_CALL_STATEMENT(
+            // If XML is default, load default plugins by absolute paths
+            _impl->register_plugins_in_registry(xmlConfigFile, xml_config_file.empty());)
```

Comment:
I think we can change this logic to load plugins from plugins.xml and after that load all other unregistered plugins from `plugins.xml`
```suggestion
    if (!xmlConfigFile.empty())
        OV_CORE_CALL_STATEMENT(
            // If XML is default, load default plugins by absolute paths
            _impl->register_plugins_in_registry(xmlConfigFile, xml_config_file.empty());)
    OV_CORE_CALL_STATEMENT(
        //  Load plugins from pre-compiled list
        _impl->register_compile_time_plugins();)

```

Reply:
Thanks for your response and yes, we are fine with that. We prepared this option by checking for already registered plugins in `register_compile_time_plugins()` such that previously loading based on the `plugins.xml` can be used to overrule the compiled-in configuration. Looking forward to receiving your comments after your discussion 👍